### DISCUSSION TITLE
unexpected instance type?

### DISCUSSION THREAD

**Discussion by Mercury-Fred** [pboettch/json-schema-validator#327] [Q&A]
Hi all,

I am attempting to use this package for json schema validation.  I've tried my schema and json file on a web page for testing such things,  (e.g. https://www.jsonschemavalidator.net/) and it all works fine.  However, when I attempt to use this package, I get "unexpected instance type".   My json file looks like the following:

{
  "property1" :
  [
     {
     "item1" : -1
     }
  ]
}

The error message includes the entire json file as the "instance" and the instance value being generated internally by the package is  the number 3.    Note that I also get the "unexpected instance type" with an empty json file... e.g. "{}"  - Then I get ERROR: '' - '"{\n}': unexpected instance type

Any suggestions?  Thanks

**Comment by pboettch** [comment]
It seems that you are not parsing your JSON-string correctly. Have you verified that your JSON-instance is correctly parsed?

**Reply by Mercury-Fred** [reply]
Thanks - yes, that was the problem.   I am now having trouble getting "unevaluatedProperties" to work as designed.  If I paste an example into other validators, e.g. https://www.jsonschemavalidator.net/, the results are as I would expect (e.g. an extra property is detected and raises an error), however, when using this package, everything succeeds.  Does the latest release support unevaluatedProperties? (I need this capability as I am using references to incorporate common properties into various derived objects.   Thanks so much for your help.

**Comment by Mercury-Fred** [comment]
Here is an example schema that doesn't seem to catch additional properties (but does in the website identified above:

    {
        "$schema": "http://json-schema.org/draft-2019-09/schema#",
        "type": "object",
        "$defs": {
            "groupsType" :  {
                                "type": "array",
                                "items": {
                                    "type": "integer",
                                    "minimum": 0,
                                    "exclusiveMaximum" : 255,
                                    "uniqueItems": true
                                }
                            },
            "groupsDef" : {
                "properties": {
                                "groups": {"$ref": "#/$defs/groupsType"},
                                "group" : {"type" : "integer",
                                        "minimum" : 0}
                            },
                "oneOf": [
                    {"allOf": [
                        { "not": {"required":["groups"]}},
                        { "not": {"required":["group"]}}
                        ]
                    },
                    {"required":["groups"]},
                    {"required":["group"]}
                ]
            }
        },
        "properties": {
            "xyz" : {
                "type" : "array",
                "items" : {
                    "type": "object",
                    "allOf": [{ "$ref": "#/$defs/groupsDef" }],
                    "properties": { "stddev": {"type": "number" }},
                    "required": [
                        "stddev"
                    ],
                    "unevaluatedProperties" : false
                }
            }
        },
        "unevaluatedProperties" : false
    }

******** Test Json file below  *********
{
   "xyz" : [
        {
            "stddev": 2.4,
            "groups": [8, 9],
          	"junk": "fred"
        },
        {
            "stddev": 321,
        }
    ],
    "fred" : { "type" : "integer"}
}

**Reply by pboettch** [reply]
Only draft-7 is supported with this library. PRs are welcome.
